oggm/tests/test_models.py

```diff
@@ -49,13 +48,12 @@
 pytest.importorskip('salem')
 
 
-def setup_module(module):
-    module._url_retrieve = utils.oggm_urlretrieve
+@pytest.fixture(autouse=True, scope='module')
+def init_url_retrieve(request):
```

Comment:
Whether this is really much better than what we had before is a bit of matter of taste. The previous (with setup and teardown) is a bit more explicit on the first sight. 

Let's keep your version, but rename to `patch_url_retrieve` and add a comment specifying that the "teardown" happens after the yield.

Reply:
While it feels essentially the same as using the `setup_module` and `teardown_module` calls, those are still very much xunit-style ways of doing things. One of the nicest things of doing this as a fixture is we only need to write it once and every test module can use it.